Merge dev branch to master

Vishwaak opened the pull request:
The dev branch is rebased properly over the master branch. Fixes issue #1  and #2

Vishwaak committed:
Compile: code cleanups

Removed the redundant code and cleanup and Refactored.

Vishwaak committed:
Compile: Add I am not done feature

Allows the user to stop the compiler execution at a certain file.

Vishwaak committed:
Compile: watch and verify commands

Helps the users to compile the file if a change is detected
and verify to check all the files.

vibhoothi reviewed (changes requested):
Please cleanup the commits,
It has lot of noise,

What I would suggest is 
- [x] Soft reset all 7
- [x] Have a single commit which explains the details what you did short and precise.

vibhoothi reviewed (commented):
Commit messages seems fine but still, it is not able to watch/verify.

<Details>

```c
vibhoothiiaanand@derpBox /V/b/m/C-lings (dev) [255]> ./test verify
problem-
���
/Volumes/buildBox/misc/C-lings/Problems/1.c
���
/Volumes/buildBox/misc/C-lings/Problems/2.c
���
/Volumes/buildBox/misc/C-lings/Problems/3.c
���
/Volumes/buildBox/misc/C-lings/Problems/a.out
clang: error: no such file or directory: '���'
clang: error: no input files
sh: line 1: /Volumes/buildBox/misc/C-lings/Problems/2.c: No such file or directory

problem not compiled,
 solve error to compile
 for hint type <problem.hint>
```
</Details>

Vishwaak commented:
There seems to be no issue with, watch/verify command in my setup. Also seems to work with clang-10 with no issue.

Vishwaak committed:
compile: Fixes runtime warnings

vibhoothi committed:
Fix runtime errors on some compilers

+ The compiler was emiting wrong symbols during runtime which was caused due to strcat, from a stackoverflow
answer
"strcat needs to find the null terminator in the destination. Your *result points to uninitialised memory, which
happens to have a null terminator characters in. Appending` \0` just before combining the strings
should fix it. Also, you are not allocating enough space for the null terminator in variables".

+ Also add some more debug logs (will be removed later)

Fixes compilation on macOS default Clang

akshayaky commented on compile.c:
```diff
@@ -3,39 +3,94 @@
     }
+    else
+    {
+        printf("Commnad not found \n");
```
Typo here

akshayaky commented on compile.c:
```diff
@@ -3,40 +3,108 @@
+         {
+            sprintf(directory,"%s", dir_list.array[j]);
+            printf("inside while %s", directory);
+	    compile(directory);
```
Not aligned.

akshayaky commented on compile.c:
```diff
@@ -3,40 +3,108 @@
+    else if(strcmp("verify",argv[1]) == 0)
+    {
+        printf("\n Inside verify %s \n",directory);
+	for(int j=1; j<exe_cnt; j++)
```
Not aligned.

akshayaky commented on compile.c:
```diff
@@ -53,7 +121,32 @@ int main()
+   char *file_name = malloc( 128 * sizeof(char));
+   FILE *fp;
+   char *str = "//I AM NOT DONE";
+    sprintf(file_name,"%s",strtok(files,"\n"));
+    printf("%s \n",files);
```
Not aligned.

akshayaky commented on compile.c:
```diff
@@ -3,40 +3,108 @@
+    }
+        sprintf(command,"%s%s%s","ls ",cwd,"/Problems/ > file.txt");
+        system(command);
+	printf("\n Total command after strcats: %s \n", command);
```
Not aligned.

akshayaky commented on compile.c:
```diff
@@ -3,40 +3,108 @@
+	sprintf(path,"%s%s",cwd,"/Problems/");
+	printf("late path: %s \n", path);
+        int j = 0;
+        while(getline(&line,&len,fp)!= -1){
+                sprintf(result.array[j], "%s%s", path, line);
+		printf("%s",result.array[j]);
+            
+            j++;       
+    }
```
- Uneven indentation.  
 - Better to use a single style of Brace Placement

akshayaky commented on compile.c:
```diff
@@ -53,7 +121,32 @@ int main()
+       }
+   }
+   if(fp)
+    {
+       fclose(fp);
+    }
```
Brace placement is different from the rest.

akshayaky reviewed (changes requested):
The lines are not properly indented and different brace placement styles are used throughout the code. Fixing these would improve Code Readability.
